library/src/main/scala/scala/scalajs/macroImpls/JSMembers.scala

```diff
@@ -0,0 +1,87 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js API               **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-lang.org/     **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+package scala.scalajs.macroImpls
+
+import Compat210._
+
+/** JSMember is an ADT more or less equivalent to Scala's MethodType
+ *  that allows to distinguish setters and getters from methods.
+ *
+ *  It also allows to check method conformance based on Scala.js' JavaScript
+ *  calling conventions.
+ *
+ *  Currently does not support polymorphic method types.
+ *
+ *  @author Tobias Schlatter
+ */
+trait JSMembers {
+  // Import macros only here, otherwise we collide with Compat210._
+  import scala.reflect.macros._
+  import blackbox.Context
+
+  val c: Context
+
+  import c.universe._
+
+  sealed trait JSMember {
+    /** Whether this JSMember conforms to that JSMember */
+    def conformsTo(that: JSMember): Boolean
+
+    /** Create a display string of this member with a given name */
+    def displayStr(name: String): String
+  }
+
+  case class JSMethodParam(info: Type, isDefault: Boolean) {
+    def conformsTo(that: JSMethodParam): Boolean =
+      (!that.isDefault || this.isDefault) && that.info <:< this.info
+
+    override def toString(): String =
+      if (isDefault) s"$info = ???"
+      else info.toString
+  }
+
+  case class JSMethod(params: List[JSMethodParam],
+      returnType: Type) extends JSMember {
+
+    def conformsTo(that: JSMember): Boolean = that match {
+      case JSMethod(ps, ret) =>
```

Comment:
Consider renaming `ps` to `thatParams` and `ret` to `thatReturnType`.
Also, consider using `resultType` instead of `returnType` (Scala methods have *results*, not *returns*).